Created new documentation of Headers.forEach() method (#19556)
mdn/content: files/en-us/web/api/headers/foreach/index.md

---
title: Headers.forEach()
slug: Web/API/Headers/forEach
page-type: web-api-instance-method
tags:
  - API
  - Fetch API
  - Headers
  - Method
  - Reference
browser-compat: api.Headers.forEach
---
{{APIRef}}

> The **`Headers.forEach()`** method executes a callback function once per each key/value pair in the {{domxref("Headers.Headers()", "Headers()")}} object.

Reviewer: ```suggestion
The **`Headers.forEach()`** method executes a callback function once per each key/value pair in the [`Headers`](/en-US/docs/Web/API/Headers) object.
```

## Syntax

```js
// Arrow function
forEach((value, key) => { /* … */ })
forEach((value, key, object) => { /* … */ })

// Inline callback function
forEach(function (value, key) { /* … */ })
forEach(function (value, key, object) { /* … */ })

> forEach(function (value, key) { /* … */ }, thisContext)

Reviewer: ```suggestion
forEach(function (value, key) { /* … */ }, thisArg)
```

Author: updated thisArg

```

### Parameters

- `callbackFn`
  - : Function to execute for each entry in the map. It takes the following arguments:

>     - `value`
>       - : Value of each iteration.
>     - `key`
>       - : Key of each iteration.

Reviewer: Possibly point out what the semantics of "key" and "value" are? e.g.

```suggestion
    - `value`
      - : Value of the currently visited header entry.
    - `key`
      - : Name of the currently visited header entry.
```

>     - `object` {{Optional_Inline}}

Reviewer: ```suggestion
    - `object`
```

I already changed it last time. `object` is _not_ optional. This is what the language passes to you, and marking it as optional means it may or may not be passed, but that's not the case. We are just ignoring this parameter.

Author: Ok, got it. thankyou.

      - : The Headers object being iterated.

> - `thisContext` {{Optional_Inline}}

Reviewer: ```suggestion
- `thisArg` {{Optional_Inline}}
```

Author: got it.

  - : Value to use as `this` when executing `callback`.

### Return value

{{jsxref("undefined")}}.

## Description

The `Headers.forEach()` method executes the provided callback once for each key of the Headers which actually exist. It is not invoked for keys which have been deleted. However, it is executed for keys which are present but have the value undefined.

## Examples

### Printing the contents of Headers object

> The following code logs a line for each key/value in Headers object.

Reviewer: ```suggestion
The following code logs a line for each key/value in the `myHeaders` object.
```

## Syntax

```js
// Arrow function
forEach((value, key)=>{ /*...*/ })
forEach((value, key, object)=>{ /*...*/ })

// Inline callback function
forEach(function(value, key){ /* ... */ })
forEach(function(value, key, object){ /* ... */ })
forEach(function(value, key){ /* ... */ }, [thisContext])
```

### Parameters

- `callbackFn`
  - : Function to execute for each entry in the map. It takes the following
    arguments:
    - `value`
      - : Value of each iteration.
    - `key`
      - : Key of each iteration.
    - `object` {{Optional_Inline}}
      - : The Headers object being iterated.
- `thisContext` {{Optional_Inline}}
  - : Value to use as `this` when executing `callback`.

### Return value

{{jsxref("undefined")}}

## Description

The `Headers.forEach()` method executes the provided callback once for each key of the Headers which actually exist. It is not invoked for keys which have been deleted. However, it is executed for keys which are present but have the value undefined.

## Examples

### Printing the contents of Headers object

The following code logs a line for each key/value in Headers object.

> 
> ```js
> // Create a new test Headers object
> const myHeaders = new Headers();
> myHeaders.append("Content-Type", "application/json")
> myHeaders.append("Cookie", "This is a demo cookie")
> myHeaders.append("compression", "gzip")
> 
> // Display the key/value pairs
> myHeaders.forEach((value, key)=> {
>     console.log(`${key} ==> ${value}`)
> })  
> ```
> 
> The result is:
> 
> ```text

Reviewer: ````suggestion

```js
// Create a new test Headers object
const myHeaders = new Headers();
myHeaders.append("Content-Type", "application/json");
myHeaders.append("Cookie", "This is a demo cookie");
myHeaders.append("compression", "gzip");

// Display the key/value pairs
myHeaders.forEach((value, key) => {
  console.log(`${key} ==> ${value}`);
})  
```

The result is:

```
````

compression ==> gzip
content-type ==> application/json
cookie ==> This is a demo cookie
```

## Browser compatibility

{{compat}}

## See also

> - [Map.forEach](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/forEach)

Reviewer: ```suggestion
- [`Map.prototype.forEach()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/forEach)
```

It occurred very weird to me that you are linking to `Map.prototype.forEach` after all. Why not `Array.p.forEach` or `NodeList` or something else?

Author: @Josh-Cena Thank you for your attention
In my sense Headers object is an interface which behaves like a Map.
For example in headers object key/value can be like :- 
  `key` = "Content-Type" , `value` = "text/html"
  `key` = "Content-Encoding" , `value` = "gzip"
or something like that.

I think `Map.prototype.forEach()` should be appropriate.

If you have other suggestions, please let me know.
I would like to get feedback on my work.
